tuning/autotune.py

```diff
@@ -167,6 +166,32 @@ def format(self, record):
     return log_file_path
 
 
+def handle_error(
+    condition: bool,
+    msg: str,
+    level: int = logging.ERROR,
+    exit_program: bool = True,
+) -> None:
+    """Handles errors with logging and optional program exit"""
+    if condition:
+        # Log the message with the specified level
```

Comment:
nit: prefer early exits to reduce nesting
```suggestion
    if not condition:
        return
    
    # Log the message with the specified level
```